### PR TITLE
Dev spring boot categories

### PR DESCRIPTION
**Pull request by OlegTuryk**

### CODE REVIEW

File: src/main/java/com/app/onlinebookstore/repository/BookRepository.java [hunk 1]
```diff
@@ -1,8 +1,11 @@
 package com.app.onlinebookstore.repository;
 
 import com.app.onlinebookstore.model.Book;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    List<Book> findAllByCategoriesId(Long categoriesId, Pageable pageable);
```
Comment (keda-vlad) [inline src/main/java/com/app/onlinebookstore/repository/BookRepository.java:12]: Check if u don't have n+1 problem without join fetch

File: src/main/resources/db/changelog/changes/07-create-categories-table.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                    nullable: false
+              - column:
+                  name: description
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (keda-vlad) [inline src/main/resources/db/changelog/changes/07-create-categories-table.yaml:29]: add empty line

File: src/main/resources/db/changelog/changes/08-create-book_category-table.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+              - column:
+                  name: book_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+              - column:
+                  name: category_id
+                  type: bigint
+                  constraints:
+                    nullable: false
```
Comment (keda-vlad) [inline src/main/resources/db/changelog/changes/08-create-book_category-table.yaml:18]: add empty line

**Review by keda-vlad — APPROVED** [review]
LGTM, only minor fixes

File: src/main/java/com/app/onlinebookstore/dto/category/CreateCategoryRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.app.onlinebookstore.dto.category;
+
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+
+@Data
+public class CreateCategoryRequestDto {
+    @NotBlank
+    private String name;
+    private String description;
```
Comment (vovasalyha) [inline src/main/java/com/app/onlinebookstore/dto/category/CreateCategoryRequestDto.java:13]: add string size restrictions

File: src/main/resources/db/changelog/changes/08-create-book_category-table.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-book_category-table
+      author: oleg
+      changes:
+        - createTable:
+            tableName: book_category
+            columns:
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/08-create-book_category-table.yaml:8]: add constraints to this table (PK, FK)

File: src/main/java/com/app/onlinebookstore/repository/BookRepository.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/com/app/onlinebookstore/repository/BookRepository.java:12]: Yes, if result of this method uses categories of a book then you'll end up with n+1, so add fetch of entity graph

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve it a bit

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**